pkg/cache/metrics.go

```diff
@@ -0,0 +1,64 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/lasso/pkg/metrics"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	defaultCacheMetricsCollectionPeriod = 1 * time.Minute
+)
+
+func (f *sharedCacheFactory) collectMetrics() sharedCacheFactoryMetrics {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+
+	gvks := make(map[schema.GroupVersionKind]int)
+	for gvk, c := range f.caches {
+		items := c.GetStore().List()
```

Comment:
I'd be curious to know if `List()` can be slow here (with Vai) such that it would be worth it to do something like:
```go
f.lock.RLock()
caches := map[gvk]cache.SharedBlabla
for gvk, c := range f.caches {
caches[gvk] = c
}
f.lock.RUnlock()

// Then to the listing with the copied caches
```

Reply:
Good point, thanks! I've followed a similar approach using `maps.Clone` to get a local copy of the map.

Reply:
I didn't know about maps.Clone, nice.

Reply:
Yeah, both [`maps`](https://pkg.go.dev/maps) and [`slices`](https://pkg.go.dev/slices) were (partially) promoted from their [experimental](https://pkg.go.dev/golang.org/x/exp) versions in recent versions of Go, which came along with the support for generics.